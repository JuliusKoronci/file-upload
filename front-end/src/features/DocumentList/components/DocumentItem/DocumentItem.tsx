import { SFC } from 'react';
import * as React from 'react';

import {
  Avatar,
  IconButton,
  ListItem,
  ListItemAvatar,
  ListItemSecondaryAction,
  ListItemText,
} from '@material-ui/core';
import { CloudDownload, Folder } from '@material-ui/icons';
import { DeleteContainer } from '../../containers/DeleteContainer';
import { IDocument, getDownloadLink } from '../../../../domains/Documents';

interface IDocumentItemProps {
  item: IDocument;
}

export const DocumentItem: SFC<IDocumentItemProps> = ({ item }) => {
  const downloadNewTab = () => window.open(getDownloadLink(item), '_blank');
  return (
    <ListItem>
      <ListItemAvatar>
        <Avatar>
          <Folder />
        </Avatar>
      </ListItemAvatar>
      <ListItemText
        primary={item.name}
        secondary={item.type}
      />
      <ListItemSecondaryAction>
        <IconButton aria-label="Download" onClick={downloadNewTab}>
          <CloudDownload />
        </IconButton>
        <DeleteContainer item={item} />
      </ListItemSecondaryAction>
    </ListItem>
  );
};