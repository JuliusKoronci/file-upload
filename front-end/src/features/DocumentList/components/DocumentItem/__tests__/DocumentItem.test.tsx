import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DocumentItem } from '../DocumentItem';


describe('DocumentItem', () => {
  it('should render snapshot', () => {
    expect(toJson(shallow(<DocumentItem
      item={{ slug: 'test', name: 'test', link: 'http://link', type: 'jpg' }}
    />))).toMatchSnapshot();
  });
});