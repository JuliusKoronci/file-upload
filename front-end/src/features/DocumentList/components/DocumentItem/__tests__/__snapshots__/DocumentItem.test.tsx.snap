// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentItem should render snapshot 1`] = `
<WithStyles(ListItem)>
  <WithStyles(ListItemAvatar)>
    <WithStyles(Avatar)>
      <pure(FolderIcon) />
    </WithStyles(Avatar)>
  </WithStyles(ListItemAvatar)>
  <WithStyles(ListItemText)
    primary="test"
    secondary="jpg"
  />
  <WithStyles(ListItemSecondaryAction)>
    <WithStyles(IconButton)
      aria-label="Download"
      onClick={[Function]}
    >
      <pure(CloudDownloadIcon) />
    </WithStyles(IconButton)>
    <withState(Connect(DeleteContainer))
      item={
        Object {
          "link": "http://link",
          "name": "test",
          "slug": "test",
          "type": "jpg",
        }
      }
    />
  </WithStyles(ListItemSecondaryAction)>
</WithStyles(ListItem)>
`;
